src/coredump/coredump.c

```diff
@@ -782,30 +784,32 @@ static int change_uid_gid(const Context *context) {
         return drop_privileges(uid, gid, 0);
 }
 
-static int setup_container_mount_tree(int mount_tree_fd, char **container_root) {
+static int attach_mount_tree(int mount_tree_fd) {
         _cleanup_free_ char *root = NULL;
         int r;
 
         assert(mount_tree_fd >= 0);
-        assert(container_root);
 
-        r = unshare(CLONE_NEWNS);
+        r = detach_mount_namespace();
         if (r < 0)
-                return log_warning_errno(errno, "Failed to unshare mount namespace: %m");
+                return log_warning_errno(r, "Failed to detach mount namespace: %m");
 
-        r = mount(NULL, "/", NULL, MS_REC|MS_PRIVATE, NULL);
-        if (r < 0)
-                return log_warning_errno(errno, "Failed to disable mount propagation: %m");
+        r = mkdir_label(MOUNT_TREE_ROOT, 0555);
```

Comment:
maybe use mkdir_p_label() here, given this is three levels down, even if /run/systemd will already exist.

You can then drop EEXIST check below.

Reply:
Not addressed? I think `mkdir_p_label()` is better.